pkg/controller/pxc/upgrade.go

```diff
@@ -536,11 +572,6 @@ func (r *ReconcilePerconaXtraDBCluster) waitPodRestart(cr *api.PerconaXtraDBClus
 				return false, errors.Wrap(err, "fetch pod")
 			}
 
-			// We update status in every loop to not wait until the end of smart update
-			if err := r.updateStatus(cr, nil); err != nil {
-				return false, errors.Wrap(err, "update status")
-			}
-
```

Comment:
Just a heads up @delgod: `status.PXC.ready` and `status.ready` will be stale during smart update.